doc/releases/migration-guide-3.5.rst

```diff
@@ -0,0 +1,36 @@
+:orphan:
+
+.. _migration_3.5:
+
+Migration guide to Zephyr 3.5.0 (Working Draft)
+###############################################
+
+This document describe the changes required or recommended when migrating your
+application to Zephyr 3.5.0.
+
+Required changes
+****************
+
+* The :kconfig:option:`CONFIG_BOOTLOADER_SRAM_SIZE` default value is now ``0`` (was
+  ``16``). Bootloaders that use a part of the SRAM should set this value to an
+  appropriate size. :github:`60371`
+
+* The Kconfig option CONFIG_GPIO_NCT38XX_INTERRUPT has been renamed to
+  :kconfig:option:`CONFIG_GPIO_NCT38XX_ALERT`.
+
+* MCUmgr SMP version 2 error codes entry has changed due to a collision with an
+  existing response in shell_mgmt. Previously, these errors had the entry ``ret``
+  but now have the entry ``err``. ``smp_add_cmd_ret()`` is now deprecated and
+  :c:func:`smp_add_cmd_err` should be used instead, ``MGMT_CB_ERROR_RET`` is
+  now deprecated and :c:enumerator:`MGMT_CB_ERROR_ERR` should be used instead.
+  SMP version 2 error code defines for in-tree modules have been updated to
+  replace the ``*_RET_RC_*`` parts with ``*_ERR_*``.
+
+Recommended Changes
+*******************
+
+* Setting the GIC architecture version by selecting
+  :kconfig:option:`CONFIG_GIC_V1`, :kconfig:option:`CONFIG_GIC_V2` and
+  :kconfig:option:`CONFIG_GIC_V3` directly in Kconfig has been deprecated.
+  The GIC version should now be specified by adding the appropriate compatible, for
```

Comment:
this reads like it is missing some descriptor to it.  maybe... `adding the appropriate device tree compatible`?  Or maybe just `adding the appropriate GIC node in the device tree compatible section, for example... `

Reply:
This was taken from the release notes so I will leave it to @carlocaione to ack this.

Reply:
The compatible is already present in the DT but previously it was a generic one `arm,gic` and then the version was specified using Kconfig. This is now deprecated and instead the version is already carried by the compatible itself.

So instead of having `arm,gic` in the DT + `CONFIG_GIC_V2` we have now only `arm,gic-v2` in the DT.

So the current version looks fine to me.